### PR TITLE
[ISSUE #29] Store the broker configuration and jvm parameters in configmap

### PR DESCRIPTION
**Pull request by zhihui921016**
Before the configuration of broker is wrote in the image, it is not convenient to modify it. It is more reasonable to store it in configmap.

### CODE REVIEW

**Comment by liuruiyiyang** [comment]
Thanks for you contribution. Could you please add some docs about the usage of the ConfigMap for the users?  :)

**Comment by vongosling** [comment]
@zhihui921016 ping, could you help to resolve the conflict and connect this pr with a concrete issue? May be "Pull Request naming conventions" in here(https://docs.google.com/document/d/1fo_Z4_nUOyykkmQOE2kEmxcLwhhFiQENQwQiy852XUM/edit#heading=h.9hxidqv8pyyk) will help you.

**Comment by zhihui921016** [comment]
@liuruiyiyang @vongosling Thanks for your suggestions. Actually I provide a demo in the example dir. You can get the usage about ConfigMap.

File: images/broker/runbroker-customize.sh [hunk 1]
```diff
@@ -130,7 +130,7 @@ Xmx=$MAX_HEAP_SIZE
 Xmn=$HEAP_NEWSIZE
 MaxDirectMemorySize=$MAX_HEAP_SIZE
 # Set for `JAVA_OPT`.
-JAVA_OPT="${JAVA_OPT} -server -Xms${Xms} -Xmx${Xmx} -Xmn${Xmn}"
+JAVA_OPT="${JAVA_OPT} -server ${BROKER_MEM:-"-Xms${Xms} -Xmx${Xmx} -Xmn${Xmn}"} "
```
Comment (liuruiyiyang) [inline images/broker/runbroker-customize.sh:133]: Could you please describe more about this line? I am not sure how it works. The original script will calculate MEM according to system memory resource. And we are going to add resource limit in future version.

Comment (zhihui921016) [reply images/broker/runbroker-customize.sh:133]: BROKER_MEM will be set as an environment variable through configmap and we can use the given value first. If BROKER_MEM does not exist, we will use the calculated value.

**Comment by liuruiyiyang** [comment]
Hi, @zhihui921016 
Since this PR has some conflicts with the master branch I am wondering if you could solve them or should a new PR be proposed.

**Comment by zhihui921016** [comment]
> Hi, @zhihui921016
> Since this PR has some conflicts with the master branch I am wondering if you could solve them or should a new PR be proposed.
Thanks for the reminder, fixed

**Comment by zhihui921016** [comment]
> Hi, @zhihui921016
> Since this PR has some conflicts with the master branch I am wondering if you could solve them or should a new PR be proposed.

Thanks for the reminder, fixed

**Review by liuruiyiyang — APPROVED** [review]
LGTM

**Pull request merged**